src/codegen.c

```diff
@@ -0,0 +1,591 @@
+                       const c_float *vecf,
+                       c_int          n,
+                       const char    *name){
+
+  c_int i;
+  if (n) {
+    fprintf(f, "c_float %s[%d] = {\n", name, n);
+    for (i = 0; i < n; i++) {
+      fprintf(f, "  (c_float)%.20f,\n", vecf[i]);
+    }
+    fprintf(f, "};\n");
+  }
+  else {
+    fprintf(f, "c_float *%s = NULL;\n", name);
+  }
+}
+
+static void write_veci(FILE        *f,
+                       const c_int *veci,
+                       c_int        n,
+                       const char  *name){
+
+  c_int i;
+  if (n) {
+    fprintf(f, "c_int %s[%d] = {\n", name, n);
+    for (i = 0; i < n; i++) {
+      fprintf(f, "  %i,\n", veci[i]);
+    }
+    fprintf(f, "};\n");
+  }
+  else {
+    fprintf(f, "c_int *%s = NULL;\n", name);
+  }
+}
+
+static void write_OSQPVectorf(FILE              *f,
+                              const OSQPVectorf *vec,
+                              const char        *name){
+  
+  char vecf_name[50];
+  sprintf(vecf_name, "%s_val", name);
+  write_vecf(f, vec->values, vec->length, vecf_name);
+  fprintf(f, "OSQPVectorf %s = {\n  %s,\n  %d\n};\n", name, vecf_name, vec->length);
+}
+
+static void write_OSQPVectori(FILE              *f,
+                              const OSQPVectori *vec,
+                              const char        *name){
+  
+  char veci_name[50];
+  sprintf(veci_name, "%s_val", name);
+  write_veci(f, vec->values, vec->length, veci_name);
+  fprintf(f, "OSQPVectori %s = {\n  %s,\n  %d\n};\n", name, veci_name, vec->length);
+}
+
+
+/*********
+* Matrix
+**********/
+
+static void write_csc(FILE       *f,
+                      const csc  *M,
+                      const char *name){
+
+char vec_name[50];
+sprintf(vec_name, "%s_p", name);
+write_veci(f, M->p, M->n+1, vec_name);
+sprintf(vec_name, "%s_i", name);
+write_veci(f, M->i, M->nzmax, vec_name);
+sprintf(vec_name, "%s_x", name);
+write_vecf(f, M->x, M->nzmax, vec_name);
+fprintf(f, "csc %s = {\n", name);
+fprintf(f, "  %d,\n", M->m);
+fprintf(f, "  %d,\n", M->n);
+fprintf(f, "  %s_p,\n", name);
+fprintf(f, "  %s_i,\n", name);
+fprintf(f, "  %s_x,\n", name);
+fprintf(f, "  %d,\n", M->nzmax);
+fprintf(f, "  %d,\n", M->nz);
+fprintf(f, "};\n");
+}
+
+static void write_OSQPMatrix(FILE             *f,
+                             const OSQPMatrix *mat,
+                             const char       *name){
+
+  char csc_name[50];
+  sprintf(csc_name, "%s_csc", name);
+  write_csc(f, mat->csc, csc_name);
+  fprintf(f, "OSQPMatrix %s = {\n", name);
+  fprintf(f, "  &%s,\n", csc_name);
+  fprintf(f, "  %d\n", mat->symmetry);
+  fprintf(f, "};\n");
+  
+}
+
+
+/**********
+* Settings
+***********/
+
+static void write_settings(FILE               *f,
+                           const OSQPSettings *settings,
+                           const char         *prefix){
+  
+  fprintf(f, "/* Define the settings structure */\n");
+  fprintf(f, "OSQPSettings %ssettings = {\n", prefix);
```

Comment:
I think it might be cleaner/clearer if there is an underscore between the prefix and the name of the variable (i.e. `%s_settings`).

Reply:
After discussion, it will be expected the user provide an underscore in their prefix if they want to have underscores separating the prefix from the variable name.